### PR TITLE
[fix/docs]: Improve backtracking/n_queens.cpp

### PR DESCRIPTION
**Pull request by Panquesito7**
#### Description of Change

- Improve `backtracking/n_queens.cpp` with better code and documentation
- Continuation of #916

#### Checklist

- [x] Added description of change
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 🎉This is the 1000th issue/PR! 🎉 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1000"></a>

### CODE REVIEW

File: backtracking/n_queens.cpp [hunk 1]
```diff
@@ -1,63 +1,123 @@
...
+ * Utility function to print matrix
+ * @tparam n number of matrix size
+ * @param board matrix where numbers are saved
+ */
+template <size_t n>
+void printSolution(const std::array<std::array<int, n>, n> &board) {
+  std::cout << "\n";
+  for (int i = 0; i < n; i++) {
+    for (int j = 0; j < n; j++) {
+      std::cout << "" << board[i][j];
```
Comment (ayaankhan98) [inline backtracking/n_queens.cpp:36]: i think it should be like
``` 
std::cout << board[i][j] << " ";
```
isn't it ?

File: backtracking/n_queens.cpp [hunk 2]
```diff
@@ -1,63 +1,123 @@
...
+    if (isSafe<n>(board, i, col)) {
+      // Place this queen in matrix
+      board[i][col] = 1;
 
-void solveNQ(int board[N][N], int col) {
-    if (col >= N) {
-        printSolution(board);
-        return;
-    }
+      // Recur to place rest of the queens
```
Comment (ayaankhan98) [inline backtracking/n_queens.cpp:100]: A minor correction
**Recursive call** to place rest of the queens

File: backtracking/n_queens.cpp (hunk 1 above)
Comment (Panquesito7) [reply backtracking/n_queens.cpp]: Fixed.

File: backtracking/n_queens.cpp (hunk 2 above)
Comment (Panquesito7) [reply backtracking/n_queens.cpp]: Fixed, thank you.

File: backtracking/n_queens.cpp [hunk 3]
```diff
@@ -1,63 +1,123 @@
...
-        for (int j = 0; j < N; j++) cout << "" << board[i][j];
-        cout << "\n";
+#include <array>
+
+/**
+ * @namespace backtracking
+ * @brief Backtracking algorithms
+ */
+namespace backtracking {
+/**
```
Comment (kvedala) [inline backtracking/n_queens.cpp:26]: can add a member namespace: `n_queens`. This way, the functions here will be separate from other functions in the `backtracking` namespace.

Comment (Panquesito7) [reply backtracking/n_queens.cpp]: Added, thanks.

**Comment by Panquesito7** [comment]
Anything else that needs to be addressed before merge?

File: backtracking/n_queens.cpp [hunk 4]
```diff
@@ -1,63 +1,129 @@
...
+ * @file n_queens.cpp
```
Comment (kvedala) [inline backtracking/n_queens.cpp:2]:
```suggestion
 * @file
```
Better to remove the filename

Comment (Panquesito7) [reply backtracking/n_queens.cpp]: Done, removed.

**Comment by Panquesito7** [comment]
Will merge when CI passes.
Thank you for your reviews. 🙂

**Pull request merged**